x-pack/plugins/reporting/public/plugin.tsx

```diff
@@ -4,52 +4,57 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import * as Rx from 'rxjs';
+import { i18n } from '@kbn/i18n';
+import { I18nProvider } from '@kbn/i18n/react';
 import React from 'react';
 import ReactDOM from 'react-dom';
+import * as Rx from 'rxjs';
 import { catchError, filter, map, mergeMap, takeUntil } from 'rxjs/operators';
-import { i18n } from '@kbn/i18n';
+import {
+  CoreSetup,
+  CoreStart,
+  NotificationsSetup,
+  Plugin,
+  PluginInitializerContext,
+} from 'src/core/public';
 import { ManagementSetup } from 'src/plugins/management/public';
-import { CoreSetup, CoreStart, Plugin, PluginInitializerContext } from 'src/core/public';
-import { I18nProvider } from '@kbn/i18n/react';
 import { UiActionsSetup } from 'src/plugins/ui_actions/public';
-
-import { ReportListing } from './components/report_listing';
-import { getGeneralErrorToast } from './components';
-
-import { ReportingNotifierStreamHandler as StreamHandler } from './lib/stream_handler';
-import { ReportingAPIClient } from './lib/reporting_api_client';
-import { GetCsvReportPanelAction } from './panel_actions/get_csv_panel_action';
-import { csvReportingProvider } from './share_context_menu/register_csv_reporting';
-import { reportingPDFPNGProvider } from './share_context_menu/register_pdf_png_reporting';
-
-import { LicensingPluginSetup } from '../../licensing/public';
+import { JobId, JobStatusBuckets } from '../';
 import { CONTEXT_MENU_TRIGGER } from '../../../../src/plugins/embeddable/public';
-import { SharePluginSetup } from '../../../../src/plugins/share/public';
-
 import {
   FeatureCatalogueCategory,
   HomePublicPluginSetup,
 } from '../../../../src/plugins/home/public';
+import { SharePluginSetup } from '../../../../src/plugins/share/public';
+import { ReportingConfigType } from '../../../legacy/plugins/reporting/server/config';
```

Comment:
I hope it's ok to import a file from `server` in this file, otherwise I'll have to refactor the polling config schema to a common area.